### PR TITLE
[Merged by Bors] - feat: port Data.Polynomial.Taylor

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Polynomial/Taylor.lean [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  map_add' f g := add_comp
+  map_smul' c f := by simp only [smul_eq_C_mul, C_mul_comp, RingHom.id_apply]
+#align polynomial.taylor Polynomial.taylor
+
+theorem taylor_apply : taylor r f = f.comp (X + C r) :=
+  rfl
+#align polynomial.taylor_apply Polynomial.taylor_apply
+
+@[simp]
+theorem taylor_x : taylor r X = X + C r := by simp only [taylor_apply, X_comp]
```
Comment (eric-wieser) [inline Mathlib/Data/Polynomial/Taylor.lean:49]:
```suggestion
theorem taylor_X : taylor r X = X + C r := by simp only [taylor_apply, X_comp]
```
and same for `C`

Comment (Parcly-Taxel) [reply Mathlib/Data/Polynomial/Taylor.lean]: Thanks, implemented.

File: Mathlib/Data/Polynomial/Taylor.lean [hunk 2]
```diff
@@ -132,7 +132,8 @@ theorem taylor_eval_sub {R} [CommRing R] (r : R) (f : R[X]) (s : R) :
 
 theorem taylor_injective {R} [CommRing R] (r : R) : Function.Injective (taylor r) := by
   intro f g h
-  apply_fun taylor (-r)  at h
+  -- porting note: `apply_fun (taylor (-r)) at h` fails
```
Comment (eric-wieser) [inline Mathlib/Data/Polynomial/Taylor.lean:135]:
```suggestion
  -- porting note: `apply_fun taylor (-r) at h` fails
```

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

**Comment by Parcly-Taxel** [comment]
bors merge

**Pull request closed without merging**